sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -162,6 +180,117 @@ public Duration getOperationTimeout() {
         return openTimeout;
     }
 
+    /**
+     *
+     * @return {@link Mono} of {@link TransactionCoordinator}
+     */
+    private Mono<TransactionCoordinator> createTransactionCoordinator() {
+        if (isDisposed()) {
+            return Mono.error(logger.logExceptionAsError(new IllegalStateException(String.format(
+                "Cannot create coordinator send link '%s' from a closed session.", TRANSACTION_LINK_NAME))));
+        }
+
+        if (transactionCoordinator != null) {
+            logger.verbose("Coordinator[{}]: Returning existing transaction coordinator.", TRANSACTION_LINK_NAME);
+            return Mono.just(transactionCoordinator);
+        }
+
+        synchronized (coordinatorLock) {
+            return createCoordinatorSendLink(TRANSACTION_LINK_NAME, openTimeout, retryPolicy)
+                .handle((amqpLink, sink) -> {
```

Comment:
Why are we using `handle` rather than a `.map` operation?

Reply:
This is going to create multiple instances of TransactionCoordinator. since you have createTransactionCoordinator() for every `create`, `commit`, `rollback` operation you do. I don't think this is what you intended to do; the `transactionCoordinator` variable is never set.